File: guess.clj
;; Define a function that generates a random number between 1 and 100
(defn generate-random-number []
  (rand-int 100))

;; Define a function that prompts the user to guess a number and returns their guess
(defn get-user-guess []
  (println "Guess a number between 1 and 100:")
  (read-line))

;; Define a function that compares the user's guess to the actual number and returns a message
(defn compare-guess [guess actual]
  (cond
    (< guess actual) "Too low!"
    (> guess actual) "Too high!"
    :else "You got it!"))

;; Define a function that runs the game
(defn play-game []
  (let [actual-number (generate-random-number)]
    (loop []
      (let [guess (Integer/parseInt (get-user-guess))]
        (println (compare-guess guess actual-number))
        (if (= guess actual-number)
          (println "You win!")
          (recur))))))

;; Call the play-game function to start the game
(play-game)

File: guess.rb
# This is a number guessing game in Ruby
# The program generates a random number between 1 and 100
# The user has to guess the number in as few attempts as possible

# Generate a random number between 1 and 100
number = rand(1..100)

# Initialize the number of attempts
attempts = 0

# Prompt the user to guess the number
puts "猜一个数字，它在1到100之间："

# Loop until the user guesses the number
loop do
  # Get the user's guess
  guess = gets.chomp.to_i

  # Increment the number of attempts
  attempts += 1

  # Compare the user's guess with the number
  if guess == number
    puts "恭喜你，你猜对了！你用了#{attempts}次尝试。"
    break
  elsif guess < number
    puts "不好意思，你猜小了。再试一次："
  else
    puts "不好意思，你猜大了。再试一次："
  end
end
